mec/pkg/model/types.go

```diff
@@ -51,25 +53,52 @@ func (i *ImageRef) String() string {
 	return fmt.Sprintf("%s/%s:%s", i.Hub, i.Repository, i.Tag)
 }
 
-func StringToImageRef(ref string) *ImageRef {
-	colonSplit := strings.SplitN(ref, ":", 2)
-	tag := colonSplit[1]
-	slashSplit := strings.Split(colonSplit[0], "/")
-	repo := slashSplit[len(slashSplit)-1]
-	hub := strings.Join(slashSplit[:len(slashSplit)-1], "/")
+func StringToImageRef(ref string) (result *ImageRef) {
```

Comment:
perhaps return an `error` as well?

this looks like a common function for every project that deals with images... probably it's already implemented somewhere else? perhaps `github.com/containers/image`?

Reply:
yeah I have that on my list of things to check, especially as this thing is growing :) I found `https://github.com/containers/image/blob/master/docker/reference` which looks to be what we need. I'd like to put that into the 2.1 bucket though, because it'll require changes all over the code base to use the new reference format

Reply:
I filed https://issues.redhat.com/browse/MAISTRA-2006 to track it